pkg/plugins/trigger/push.go

```diff
@@ -21,28 +21,46 @@ import (
 
 	prowapi "sigs.k8s.io/prow/pkg/apis/prowjobs/v1"
 	"sigs.k8s.io/prow/pkg/config"
+	"sigs.k8s.io/prow/pkg/git/v2"
 	"sigs.k8s.io/prow/pkg/github"
 	"sigs.k8s.io/prow/pkg/pjutil"
 )
 
-func listPushEventChanges(pe github.PushEvent) config.ChangedFilesProvider {
+func listPushEventChanges(gc git.ClientFactory, pe github.PushEvent) config.ChangedFilesProvider {
+	var changedFiles []string
```

Comment:
This should be the only actual "breaking change", ie: cache `changedFiles` just like we do in `pull-request.go`, to avoid a git client `diff` call for each configured post submit when we are in the fallback case.